bundles/org.openhab.binding.ipcamera/README.md

```diff
@@ -0,0 +1,1312 @@
+
+There are multiple ways to get a moving picture, to use them just enter the url into any browser using ``http://192.168.xxx.xxx:SERVER_PORT/name.format`` replacing the name.format with one of the options that are listed below:
+
++ **ipcamera.m3u8** HLS (Http Live Streaming) which uses h264 compression. 
+This can be used to cast to Chromecast devices, or can display video in many browsers (some browsers require a plugin to be installed). Please understand that this format due to the way it works will give you lag behind realtime, more on this below.
++ **ipcamera.mjpeg** whilst needing more bandwidth, it is far more compatible for displaying in a wider range of UI's and browsers. It is normally 1 second or less behind real-time.
+Ffmpeg can be used to create this stream if your camera does not create one for you, but this uses more CPU. 
+A lot of cameras limit the resolution in this format, so consider using HLS, autofps.mjpeg, or snapshots.mjpeg instead which will be in a higher resolution.
++ **snapshots.mjpeg** is a special mjpeg stream created from the cameras snapshots that are taken at the polling rate.
++ **autofps.mjpeg** This requires a camera that has a motion alarm to be turned on or it will only send a picture every 8 seconds. You can also use the ``externalMotion`` channel to change the framerate.
+This feature is designed to keep data traffic to your mobile devices as low as possible by automatically sending 1fps when motion is occurring, but only 1 picture every 8 seconds when the picture has no motion.
+The idea is to not send lots of pictures if the picture has not changed as doing so only eats up your data plan.
++ **ipcamera.gif** This is small in size and very compatible and handy to use in push notifications, Pushover, Telegram, or emails. You can cast it which can be handy to show a moving picture that keeps repeating on a Google/Nest home hub or your wall mounted tablet. 
++ MP4 recordings can be created by the binding and FFmpeg, more on this below.
+
+This forum thread has examples of how to use snapshots and streams in a sitemap.
+<https://community.openhab.org/t/ip-camera-how-to-clickable-thumbnail-overview-in-sitemaps-that-opens-up-to-a-larger-view/77990>
+
+If you use Habpanel, then these widgets are worth checking out.
+<https://community.openhab.org/t/custom-widget-camera-clickable-thumbnails-that-open-a-stream/101275>
+
+
+
+## MP4 Recordings
+
+The binding can use FFmpeg to create a recording to a file.
+To do this:
+
++ Consider setting the String channel that is called `mp4Filename` to a date and time stamp in a format that you like, or leave the channel empty for the filename to default to `ipcamera.mp4`.
++ Change the Number channel called `recordMp4` to a number of how many seconds that you wish to record for. The recording will then start.
```

Comment:
This might be a candidate for an Action, too.

Reply:
What would be the benefit for changing? I have not looked at actions before so this is a genuine question.

Reply:
It's more a modeling thing. Channels are supposed to represent values. Actions can be used to send commands to Things.

Reply:
OK so from looking at this trying to learn and understand, what your suggesting is to move the COMMAND part of the channel to an ACTION, and then my guess is I would need to keep the channel for the the STATUS part that flags when the recorded file is completed? I was toying with the idea of extending this feature to count down the seconds left until the recording finished with 0 being the file is ready for use. This data could be used in rules to make decisions. The channels name would need to be changed to reflect its new use.
@Hilbrand 
Unless I get help with this I doubt I can get this change done before 2.5.9 due to my need to learn a lot before making such a change. If someone was to create an empty function/class with short comments of what to add where, I'm sure I could make the changes moving things around quickly without them needing to learn/understand my code. Since you offered to help before, I have given a shout here.

Reply:
I just wanted you to recommend [Command Descriptions](https://www.openhab.org/docs/developer/bindings/thing-xml.html#command-description) but these have fixed commands, which doesn't work with your variable setting of the filename and duration.

As you need two Channels for calling *one* function, Actions would even fit better. You could specify the filename as the first argument and the duration as second. It's really not that hard to implement an action. Here is an example: https://github.com/openhab/openhab-addons/pull/8250/files#diff-6c1fdb2ea72cc5c312076899bca70867 Basically you need a class implementing the `ThingActions` interface, a Java method annotated with `@RuleAction` and a static method. These are called when the user invokes the Action.

Reply:
Thankfully this doc helps as does your linked example.
https://www.openhab.org/docs/developer/bindings/#actions-bound-to-a-thing
Fingers crossed it triggers my test code without too much issue first go as this kind of thing I have no idea how the framework and the logic flows to know how to fault find.

Reply:
OK I have it triggering the action, so its just a matter of moving the code over and then a huge re-write of the readme which will be interesting since I don't understand this fully...

In a rule I trigger the recording like this, is there a better way? Hopefully NGRE makes this far easier...
getActions("ipcamera", "ipcamera:hikvision:BackyardCam").recordMP4("foo",10)

Reply:
DONE.